modules/REST/V1/Settings.php

```diff
@@ -103,10 +103,21 @@ public function update_item( $request ) {
 		);
 
 		foreach ( $options as $option_name => $new_value ) {
-			$result = $this->options->set( $option_name, $new_value );
+			$previous_value = $this->options->get( $option_name );
+			$result         = $this->options->set( $option_name, $new_value );
 
 			if ( $result->has_errors() ) {
 				$errors->merge_from( $result );
+				continue;
+			}
+
+			if ( $this->options->get( $option_name ) === $previous_value ) {
+				continue;
+			}
+
+			$options_with_flush = array( 'rewrite', 'force_lang', 'hide_default' );
+			if ( in_array( $option_name, $options_with_flush, true ) ) {
+				flush_rewrite_rules();
 			}
```

Comment:
What do you think of?
```suggestion
			switch ( $option_name ) {
				case 'rewrite':
				case 'force_lang':
				case 'hide_default':
					flush_rewrite_rules();
			}
```

Reply:
I like it!